internal/status/client.go

```diff
@@ -0,0 +1,28 @@
+package status
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+)
+
+type StatusClient interface {
+	GetPodsComponent(ctx context.Context, componentName string, stack v1alpha1.Microservices) (*corev1.PodList, error)
+	PatchStatus(ctx context.Context, changed, original *v1alpha1.Microservices) error
+}
+
+type StatusClientStub struct {
```

Comment:
I'm curious: is there any special reason why this is suffixed as `Stub`? I'm asking because Stubs are a kind of Mocks (not exactly) and I would expect that something suffixed as `Stub` is related to testing

Reply:
It is only for tests, mm  🤔  this should not  be exposed outside the package

Reply:
Lets move the object into `client_test.go`

Reply:
Done